nexus/operation.go

```diff
@@ -36,6 +40,16 @@ func (r operationReference[I, O]) Name() string {
 	return string(r)
 }
 
+func (operationReference[I, O]) InputType() reflect.Type {
+	var zero [0]I
+	return reflect.TypeOf(zero).Elem()
```

Comment:
```suggestion
	return reflect.TypeOf((*I)(nil)).Elem()
```

This may also work (untested), but what you have is fine/cheap too, no need to change.